components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeService.java

```diff
@@ -306,6 +324,350 @@ public Scope updateScope(Scope updatedScope) throws IdentityOAuth2ScopeException
+        String appId = resolveAppIdFromAppName(appName, tenantId);
+        try {
+            UserApplicationScopeConsentDO updatedUserApplicationScopeConsents =
+                    new UserApplicationScopeConsentDO(appId, approvedScopes, disapprovedScopes);
+            OAuthTokenPersistenceFactory.getInstance().getOAuthUserConsentedScopesDAO()
+                    .updateExistingConsentForApplication(userId, tenantId, updatedUserApplicationScopeConsents);
+            if (log.isDebugEnabled()) {
+                log.debug("Successfully updated the user consent for OAuth scopes for user : " + userId +
+                        " and application : " + appName + " in tenant with Id : " + tenantId);
+            }
+        } catch (IdentityOAuth2ScopeConsentException e) {
+            String msg = String.format(ERROR_CODE_FAILED_TO_UPDATE_USER_CONSENT_FOR_APP.getMessage(), userId, appName,
+                    tenantId);
+            if (log.isDebugEnabled()) {
+                log.debug(msg, e);
+            }
+            throw new IdentityOAuth2ScopeConsentException(
+                    ERROR_CODE_FAILED_TO_UPDATE_USER_CONSENT_FOR_APP.getCode(), msg, e);
+        }
+    }
+
+    /**
+     * Revoke scope consents of a given user for a given application.
+     *
+     * @param user      Authenticated user.
+     * @param appName   Application identifier.
+     * @param tenantId  Tenant Id.
+     * @throws IdentityOAuth2ScopeConsentException
+     */
+    public void revokeUserConsentForApplication(AuthenticatedUser user, String appName, int tenantId)
+            throws IdentityOAuth2ScopeConsentException {
+
+        String userId = resolveUserIdFromAuthenticatedUser(user);
+        String appId = resolveAppIdFromAppName(appName, tenantId);
+        try {
+            OAuthTokenPersistenceFactory.getInstance().getOAuthUserConsentedScopesDAO()
+                    .deleteUserConsentOfApplication(userId, appId, tenantId);
+            if (log.isDebugEnabled()) {
+                log.debug("Successfully revoked the user consents for OAuth scopes for user : " + userId +
+                        " and application : " + appName + " for tenant with Id : " + tenantId);
+            }
+        } catch (IdentityOAuth2ScopeConsentException e) {
+            String msg = String.format(ERROR_CODE_FAILED_TO_REVOKE_USER_CONSENT_FOR_APP.getMessage(), userId, appName,
+                    tenantId);
+            if (log.isDebugEnabled()) {
+                log.debug(msg, e);
+            }
+            throw new IdentityOAuth2ScopeConsentException(
+                    ERROR_CODE_FAILED_TO_REVOKE_USER_CONSENT_FOR_APP.getCode(), msg, e);
+        }
+    }
+
+    /**
+     * Revoke all scope consents for the user.
+     *
+     * @param user      Authenticated user.
+     * @param tenantId  Tenant Id.
+     * @throws IdentityOAuth2ScopeConsentException
+     */
+    public void revokeUserConsents(AuthenticatedUser user, int tenantId)
```

Comment:
Can we have userId as the parameter instead of AuthennticatedUser?

Reply:
Fixed